src/status_im2/subs/chat/messages.cljs

```diff
@@ -112,16 +112,32 @@
   [messages]
   (get
    (reduce (fn [{:keys [messages albums]} message]
-             (let [album-id (:album-id message)
-                   albums   (cond-> albums album-id (update album-id conj message))
-                   messages (if album-id
-                              (conj (filterv #(not= album-id (:album-id %)) messages)
-                                    {:album        (get albums album-id)
-                                     :album-id     album-id
-                                     :albumize?    (:albumize? message)
-                                     :message-id   album-id
-                                     :content-type constants/content-type-album})
-                              (conj messages message))]
+             (let [album-id  (:album-id message)
```

Comment:
This subscription is pretty heavy in logic, but I only see one test with one assertion. Could you add a couple more tests for this subscription? The reasoning is simple, if another contributor wants to change this subscription, that person will need to manually retest all the conditions, and this is no joke for the amount of things going on here.

Reply:
You are right. I have updated the test.

Reply:
Thank you! :flying_saucer: :rocket: